src/main/java/ch/njol/skript/expressions/ExprExperience.java

```diff
@@ -43,76 +45,88 @@
 	
 	@Override
 	@Nullable
-	protected Experience[] get(final Event e) {
-		if (e instanceof ExperienceSpawnEvent)
-			return new Experience[] {new Experience(((ExperienceSpawnEvent) e).getSpawnedXP())};
-		else if (e instanceof BlockBreakEvent)
-			return new Experience[] {new Experience(((BlockBreakEvent) e).getExpToDrop())};
-		else if (e instanceof PlayerExpChangeEvent)
-			return new Experience[] {new Experience(((PlayerExpChangeEvent) e).getAmount())};
+	protected Experience[] get(Event event) {
+		if (event instanceof ExperienceSpawnEvent)
+			return new Experience[] {new Experience(((ExperienceSpawnEvent) event).getSpawnedXP())};
+		else if (event instanceof BlockBreakEvent)
+			return new Experience[] {new Experience(((BlockBreakEvent) event).getExpToDrop())};
+		else if (event instanceof PlayerExpChangeEvent)
+			return new Experience[] {new Experience(((PlayerExpChangeEvent) event).getAmount())};
+		else if (event instanceof EntityBreedEvent)
+			return new Experience[] {new Experience(((EntityBreedEvent) event).getExperience())};
 		else
 			return new Experience[0];
 	}
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		switch (mode) {
-			case ADD:
-			case DELETE:
-			case REMOVE:
-			case REMOVE_ALL:
-				return new Class[] {Experience[].class, Number[].class};
 			case SET:
-				return new Class[] {Experience.class, Number.class};
+			case DELETE:
 			case RESET:
-				return null;
+				return CollectionUtils.array(ExprExperience.class, Integer.class);
+			case ADD:
+			case REMOVE:
+				return CollectionUtils.array(ExprExperience[].class, Integer[].class);
 		}
 		return null;
 	}
 	
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		double eventExp;
-		if (e instanceof ExperienceSpawnEvent) {
-			eventExp = ((ExperienceSpawnEvent) e).getSpawnedXP();
-		} else if (e instanceof BlockBreakEvent) {
-			eventExp = ((BlockBreakEvent) e).getExpToDrop();
-		} else if (e instanceof PlayerExpChangeEvent) {
-			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		int eventExp;
+		if (event instanceof ExperienceSpawnEvent) {
+			eventExp = ((ExperienceSpawnEvent) event).getSpawnedXP();
+		} else if (event instanceof BlockBreakEvent) {
+			eventExp = ((BlockBreakEvent) event).getExpToDrop();
+		} else if (event instanceof PlayerExpChangeEvent) {
+			eventExp = ((PlayerExpChangeEvent) event).getAmount();
+		} else if (event instanceof EntityBreedEvent) {
+			eventExp = ((EntityBreedEvent) event).getExperience();
 		} else {
 			return;
 		}
+
 		if (delta == null) {
 			eventExp = 0;
 		} else {
 			for (Object obj : delta) {
-				double value = obj instanceof Experience ? ((Experience) obj).getXP() : ((Number) obj).doubleValue();
+				int value = obj instanceof Experience ? ((Experience) obj).getXP() : (Integer) obj;
```

Comment:
Shouldn't be changing from double to int here. There are other events that support a double in this class. Cast the double to an int when you set it in your added events.

Reply:
The only experience method that doesn't use int from what Ayham said is the experience for cooking recipes, as such I'm unsure whether or not that should be given it's own class or if it should be given priority and added to this class.

After some more digging around I can safely say this is the only class that would do that, all other experience related systems return an `int`